tools/wpt/revlist.py

```diff
@@ -0,0 +1,299 @@
+        "help": "Regular interval of seconds used to get the tagged revision. "
+                "three-hourly, six-hourly, eight-hourly, twelve-hourly, "
+                "daily, weekly or 3600, 7200, 10800 ...",
+        "parser": parser_epoch_step
+    },
+    "head_revision": {
+        "name": "head-revision",
+        "default": DEFAULT_HEAD_REVISION,
+        "help": "Head revision (default: %s)" % DEFAULT_HEAD_REVISION,
+        "parser": lambda **kwargs: kwargs.get("head_revision",
+                                              DEFAULT_HEAD_REVISION),
+    },
+    "log_level": {
+        "name": "log-level",
+        "default": DEFAULT_LOG_LEVEL,
+        "help": "Debug log level. For example: DEBUG, INFO, WARNING, ERROR, "
+                "CRITICAL or an integer value like 10, 20 ",
+        "parser": parser_debug_log_level
+    },
+    "num_revisions": {
+        "name": "num-revisions",
+        "default": DEFAULT_NUM_REVISIONS,
+        "help": "Number of revisions to get.",
+        "parser": parser_num_revisions
+    },
+    "verbose": {
+        "name": "verbose",
+        "default": DEFAULT_VERBOSE,
+        "help": "Show more information in the output",
+        "parser": lambda **kwargs: kwargs.get("verbose", DEFAULT_VERBOSE),
+        "type": bool
+    }
+}
+
+
+def get_git_cmd(repo_path):
+    # type: (bytes) -> Callable[..., Text]
+    """Create a function for invoking git commands as a subprocess."""
+    def git(cmd, *args):
+        # type: (Text, *Union[bytes, Text]) -> Text
+        arg_list = list(item.decode("utf8") if isinstance(item, bytes) else item for item in args)  # noqa E501
+        full_cmd = [u"git", cmd] + arg_list  # type: List[Text]
+        try:
+            logger.debug(" ".join(full_cmd))
+            return subprocess.check_output(full_cmd, cwd=repo_path).decode("utf8").strip()  # noqa E501
+        except subprocess.CalledProcessError as e:
+            logger.critical("Git command exited with status %i" % e.returncode)
+            logger.critical(e.output)
+            sys.exit(1)
+    return git
+
+
+def get_tagged_revisions(commit, min_age, skip=0, max_count=10):
+    # type: (bytes, int, int, int) -> Dict[..., Dict]
+    '''
+    Returns the tagged revisions indexed by the committer date.
+    '''
+    git = get_git_cmd(wpt_root)
+    command = [
+        "--no-walk",
+        "--skip=%s" % skip,
+        "--max-count=%s" % max_count,
+        "--min-age=%s" % min_age,
+        "--format=epoch:%ct:commit:%H:%D",
+        commit
+    ]
+    result = {}
+    for output_line in git("rev-list", *command).split("\n")[1::2]:
+        # print output_line
+        output = output_line.split(":")
+        if len(output) > 5 and output[5].find(TAGGED_REVISIONS) > -1:
+            revision = {}
+            revision["epoch"] = int(output[1])
+            revision["commit"] = output[3].strip()
+            revision["tag"] = output[5].strip()
+            result[int(output[1])] = revision
```

Comment:
`./wpt rev-list --num-revisions 100` is now fast enough locally for me, but it seems like indexing by the time isn't necessary here. There's no lookup using the time, just filtering based on it, so returning a tuple `(tag, commit, date)` would be clearer I think. If the list is sorted by committerdate it should also make an optimization of `get_newer_tagged_revision` very easy, although I don't think that's necessary.